addons/binding/org.openhab.binding.innogysmarthome/src/main/java/org/openhab/binding/innogysmarthome/handler/InnogyDeviceHandler.java

```diff
@@ -0,0 +1,996 @@
+                                c.getCapabilityState().getTwoWayMeterEnergyConsumptionSensorEnergyPerMonthInEuroState(),
+                                c);
+                        updateStateForEnergyChannel(CHANNEL_ENERGY_DAY_KWH,
+                                c.getCapabilityState().getTwoWayMeterEnergyConsumptionSensorEnergyPerDayInKWhState(),
+                                c);
+                        break;
+                    case Capability.TYPE_TWOWAYMETERENERGYFEEDSENSOR:
+                        updateStateForEnergyChannel(CHANNEL_ENERGY_FEED_MONTH_KWH,
+                                c.getCapabilityState().getTwoWayMeterEnergyFeedSensorEnergyPerMonthInKWhState(), c);
+                        updateStateForEnergyChannel(CHANNEL_TOTAL_ENERGY_FED,
+                                c.getCapabilityState().getTwoWayMeterEnergyFeedSensorTotalEnergyState(), c);
+                        updateStateForEnergyChannel(CHANNEL_ENERGY_FEED_MONTH_EURO,
+                                c.getCapabilityState().getTwoWayMeterEnergyFeedSensorEnergyPerMonthInEuroState(), c);
+                        updateStateForEnergyChannel(CHANNEL_ENERGY_FEED_DAY_EURO,
+                                c.getCapabilityState().getTwoWayMeterEnergyFeedSensorEnergyPerDayInEuroState(), c);
+                        updateStateForEnergyChannel(CHANNEL_ENERGY_FEED_DAY_KWH,
+                                c.getCapabilityState().getTwoWayMeterEnergyFeedSensorEnergyPerDayInKWhState(), c);
+                        break;
+                    case Capability.TYPE_TWOWAYMETERPOWERCONSUMPTIONSENSOR:
+                        updateStateForEnergyChannel(CHANNEL_POWER_WATT,
+                                c.getCapabilityState().getTwoWayMeterPowerConsumptionSensorPowerInWattState(), c);
+                        break;
+                    default:
+                        logger.debug("Unsupported capability type {}.", c.getType());
+                        break;
+                }
+
+            }
+        } else {
+            logger.trace("DeviceId {} not relevant for this handler (responsible for id {})", device.getId(), deviceId);
+        }
+    }
+
+    /**
+     * Updates the state for the {@link Channel} of an energy {@link Device}.
+     *
+     * @param channelId
+     * @param state
+     * @param capability
+     */
+    private void updateStateForEnergyChannel(String channelId, Double state, Capability capability) {
+        if (state != null) {
+            DecimalType newValue = new DecimalType(state);
+            updateState(channelId, newValue);
+        } else {
+            logger.debug("State for {} is STILL NULL!! cstate-id: {}, c-id: {}", capability.getType(),
+                    capability.getCapabilityState().getId(), capability.getId());
+        }
+    }
+
+    @Override
+    public synchronized void onDeviceStateChanged(Device device, Event event) {
+        if (deviceId.equals(device.getId())) {
```

Comment:
here again you might return early with the logging from line 946 and do the happy path with one indention level less.